components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SubscriptionValidationSQLConstants.java

```diff
@@ -348,161 +366,256 @@
-                    "    LEFT JOIN" +
-                    "        AM_API_RESOURCE_SCOPE_MAPPING SCOPE " +
-                    "            ON URL.URL_MAPPING_ID = SCOPE.URL_MAPPING_ID " +
-                    "    WHERE" +
-                    "        URL.URL_MAPPING_ID = PROD.URL_MAPPING_ID " +
-                    "        AND API.API_ID = PROD.API_ID " +
-                    "        AND API.API_VERSION = ? " +
-                    "        AND API.CONTEXT = ?";
+            "SELECT " +
+                    "   APIS.API_ID," +
+                    "   APIS.API_PROVIDER," +
+                    "   APIS.API_NAME," +
+                    "   APIS.API_TIER," +
+                    "   APIS.API_VERSION," +
+                    "   APIS.CONTEXT," +
+                    "   APIS.API_TYPE," +
+                    "   APIS.URL_MAPPING_ID," +
+                    "   APIS.HTTP_METHOD," +
+                    "   APIS.AUTH_SCHEME," +
+                    "   APIS.URL_PATTERN," +
+                    "   APIS.RES_TIER," +
+                    "   APIS.SCOPE_NAME," +
+                    "   DEF.DEFAULT_API_VERSION " +
+                    "FROM " +
+                    "   (" +
+                    "      SELECT " +
+                    "         API.API_ID," +
+                    "         API.API_PROVIDER," +
+                    "         API.API_NAME," +
+                    "         API.API_TIER," +
+                    "         API.API_VERSION," +
+                    "         API.CONTEXT," +
+                    "         API.API_TYPE," +
+                    "         URL.URL_MAPPING_ID," +
+                    "         URL.HTTP_METHOD," +
+                    "         URL.AUTH_SCHEME," +
+                    "         URL.URL_PATTERN," +
+                    "         URL.THROTTLING_TIER AS RES_TIER," +
+                    "         SCOPE.SCOPE_NAME" +
+                    "      FROM " +
+                    "         AM_API API," +
+                    "         AM_API_PRODUCT_MAPPING PROD," +
+                    "         AM_API_URL_MAPPING URL" +
+                    "         LEFT JOIN " +
+                    "            AM_API_RESOURCE_SCOPE_MAPPING SCOPE" +
+                    "            ON URL.URL_MAPPING_ID = SCOPE.URL_MAPPING_ID" +
+                    "      WHERE " +
+                    "         URL.URL_MAPPING_ID = PROD.URL_MAPPING_ID" +
+                    "         AND API.API_ID = PROD.API_ID" +
+                    "         AND API.API_VERSION = ?" +
+                    "         AND API.CONTEXT = ?" +
+                    "   )" +
+                    "   AS APIS " +
+                    "   LEFT JOIN " +
+                    "      AM_API_DEFAULT_VERSION AS DEF " +
+                    "      ON APIS.API_NAME = DEF.API_NAME" +
+                    "      AND APIS.API_PROVIDER = DEF.API_PROVIDER" +
+                    "      AND APIS.API_VERSION = DEF.DEFAULT_API_VERSION";
```

Comment:
@tharikaGitHub don't we hv to get the PUBLISHED_DEFAULT_API_VERSION? shall we check this